swri_dbw_interface/doc/package.md

```diff
@@ -0,0 +1,366 @@
+the "/vehicle_interface" namespace.
+
+## Status Feedback [⬆](#swri_dbw_interface)
+
+Most axes/devices have a status feedback topic that is published at
+SLOW/ONCHANGE.  In all cases, the status field should follow the
+convention described in the HealthStatus message definition, and the
+message field should be filled out with an informative message
+describing the most critical issue.
+
+Some systems provide specific health feedback for critical devices
+like steering, throttle, or brake, which can be communicated using the
+status feedback.  Otherwise, the status feedback will typically be
+determined by whether or not the interface node is communicating with
+the DBW hardware.
+
+## Steering Interface [⬆](#swri_dbw_interface)
+
+The steering interface has two main topics for control:
+
+- **/vehicle_interface/steering_command** -
+*(marti_common_msgs/Float64Stamped)* - *subscription* - Desired
+position of the steering axis normalized between [0.0, 1.0].  Expected
+at a constant rate from the high level ROS system (typically 50Hz).
+
+- **/vehicle_interface/steering_feedback** -
+*(marti_common_msgs/Float64Stamped)* - *publication* - Measured
+position of the steering axis normalized between [0.0, 1.0].  Expected
+to be published at a constant rate to the rest of the system
+(typically 50Hz).
+
+The steering axis is typically normalized between the documented range
+of the underlying DBW system (e.g. [0,1023]).  The direction does not
+matter (0.0 may map to full left or full right) and is accounted for
+by the steering calibration.
+
+In some cases, we have encountered platforms that only use a tiny
+portion of the documented range (e.g, a system with a documented range
+of [0, 65535] with a useful travel only over [4000, 8000].  In those
+cases, it is handy to only map slightly more than than the used range
+(e.g. [3500 - 8500]) instead of the documented range for convenient
+use with the steering calibration tool.
+
+It is extremely helpful for the steering feedback to be provided even
+in manual mode for calibration and continuity when switching into
+robotic mode.  Every effort should be made to get a DBW supplier to
+support this.  If not, the steering has to be calibrated under robotic
+control using joystick for control which is much less convenient.
+
+- **/vehicle_interface/steering_status** -
+*(marti_common_msgs/HealthStatus)* - *publication* - Summarized health status of the
+steering system.  The status field should follow the guidelines in the
+HealthStatus mesage definition, and the message field should be filled
+out with an informative value. SLOW/ONCHANGE.
+
+
+## Throttle Interface [⬆](#swri_dbw_interface)
+
+The throttle interface has two main topics for control:
+
+- **/vehicle_interface/throttle_command** -
+*(marti_common_msgs/Float64Stamped)* - *subscription* - Desired
+position of the throttle axis normalized between [0.0, 1.0].  Expected
+at a constant rate from the high level ROS system (typically 50Hz).
```

Comment:
I still think it's a good idea to not be wishy washy on the rates here.  Specifying a rate (and perhaps tolerances) facilitates routine monitoring of health. 

The same goes for timeouts. 

Any vehicles for which the standard values don't work, can be customized, and treated as an exception to the standard.

Reply:
There's not an obvious place to put this comment, but it vaguely relates to my previous comment about the rate and timeout monitoring...

I think there should be some discussion on error handling and/or failure modes. For example, what should happen if a steering command times out?  This is not a simple modification, so it's probably fine to merge this before addressing this issue, but I think it's important to include.